nbgrader/server_extensions/formgrader/static/js/gradebook_notebook_submissions.js

```diff
@@ -187,6 +189,33 @@ var loadSubmittedNotebooks = function () {
     });
 };
 
+// at least one hidden name -> label="Show All Names"
+// no hidden name -> label="Hide All Names"
+let updateToggleNamesButtonLabel = function() {
+    const button = document.getElementById("toggle_names");
+    const icons = document.getElementsByClassName("glyphicon name-hidden");
+    const icon_array = [...icons];
+    if (icon_array.some(el => el.checkVisibility())){
```

Comment:
`checkVisibility()` might not be the best option because it is [not supported in safari](https://caniuse.com/mdn-api_element_checkvisibility). I can look for an alternative if that is a concern.

Reply:
The playwright browsers seemed too old for this anyway, so I changed it to `$(el).is(':visible')`.